addons/hw_drivers/websocket_client.py

```diff
@@ -75,7 +80,24 @@ def __init__(self, url):
     def run(self):
         self.ws = websocket.WebSocketApp(self.url,
             on_open=self.on_open, on_message=on_message,
-            on_error=on_error)
-        while 1: # A loop is necessary because reconnection must occur in all cases, even if the server closes the connection properly
-            self.ws.run_forever()
-            time.sleep(10) # Wait 10 second between each reconnection attempts
+            on_error=on_error, on_close=on_close)
+
+        # The IoT synchronised servers can stop in 2 ways that we need to handle:
+        #  A. Gracefully:
+        #   In this case a disconnection signal is sent to the IoT-box
+        #   The websocket is properly closed, but it needs to be established a new connection when
+        #   the server will be back.
+        #
+        # B. Forced/killed:
+        #   In this case there is no disconnection signal received
+        #
+        #   This will also happen with the graceful quit as `reconnect` will trigger if the server
+        #   is offline while attempting the new connection
+        while True:
+            try:
+                run_res = self.ws.run_forever(reconnect=10)
+                _logger.debug("websocket run_forever return with %s", run_res)
+            except Exception:
+                _logger.exception("An unexpected exception happened when running the websocket")
+            _logger.debug('websocket will try to restart in 10 seconds')
```

Comment:
seems unnecessary as we will either get a logger.exception or another logger.debug after this, but feel free to ignore